be/src/storage/tablet_updates.cpp

```diff
@@ -234,41 +234,31 @@ Status TabletUpdates::_load_from_pb(const TabletUpdatesPB& tablet_updates_pb) {
     // Load delete vectors and update RowsetStats.
     // TODO: save num_dels in rowset meta.
 
-    std::map<uint32_t, std::vector<std::pair<int32_t, DelVectorPtr>>> del_vector_by_rsid;
-    uint32_t max_rsid = 0;
-    uint32_t min_rsid = 1 << 31;
+    std::unordered_map<uint32_t, ssize_t> del_vector_cardinality_by_rssid;
     for (auto& [rsid, rowset] : _rowsets) {
         if (unapplied_rowsets.find(rsid) == unapplied_rowsets.end()) {
-            std::vector<std::pair<int32_t, DelVectorPtr>> del_vectors;
-            del_vectors.resize(rowset->num_segments(), {0, nullptr});
-            del_vector_by_rsid[rsid] = del_vectors;
-            if (max_rsid <= rsid) {
-                max_rsid = rsid + rowset->num_segments();
-            }
-            if (min_rsid >= rsid) {
-                min_rsid = rsid;
+            for (uint32_t i = 0; i < rowset->num_segments(); i++) {
+                del_vector_cardinality_by_rssid[rsid + i] = -1;
             }
         }
     }
 
     RETURN_IF_ERROR(TabletMetaManager::del_vector_iterate(
-            _tablet.data_dir()->get_meta(), _tablet.tablet_id(), min_rsid, max_rsid,
+            _tablet.data_dir()->get_meta(), _tablet.tablet_id(), 0, UINT32_MAX,
             [&](uint32_t segment_id, int64_t version, std::string_view value) -> bool {
-                auto iter = del_vector_by_rsid.lower_bound(segment_id);
-                if (iter == del_vector_by_rsid.end()) {
+                auto iter = del_vector_cardinality_by_rssid.find(segment_id);
+                if (iter == del_vector_cardinality_by_rssid.end()) {
                     std::string msg = strings::Substitute("found delete vector of non exist rowset");
                     LOG(ERROR) << msg;
+                    DCHECK(false);
```

Comment:
The `del_vector_cardinality_by_rssid` is generated by `_rowsets` which is only contains active rowsets. So they will be some delvecs not included by `del_vector_cardinality_by_rssid`?

Reply:
ok